File: VidPlayerlearn/app/src/main/java/com/example/vidplayerlearn/MainActivity.java
package com.example.vidplayerlearn;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.VideoView;

public class MainActivity extends AppCompatActivity {
    VideoView vid;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        vid = findViewById(R.id.vid);
        String path = "android.resource://" + getPackageName() + "/raw/videoplayback";
        Uri uri = Uri.parse(path);
        vid.setVideoURI(uri);
        vid.start();

        MediaController mc = new MediaController(this);
        vid.setMediaController(mc);
        mc.setAnchorView(vid);
    }
}

File: Day29/app/src/main/java/com/example/day29/MainActivity.kt
package com.example.day29

import android.media.MediaPlayer
import android.os.Bundle
import android.widget.ImageView
import android.widget.SeekBar
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    private lateinit var mediaPlayer: MediaPlayer
    private var totalTime: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val playbutton = findViewById<ImageView>(R.id.play)
        val pausebutton = findViewById<ImageView>(R.id.pause)
        val stopbutton = findViewById<ImageView>(R.id.stop)
        val seekbar = findViewById<SeekBar>(R.id.seekbar)

        mediaPlayer = MediaPlayer.create(this, R.raw.music)
        mediaPlayer.setVolume(1f, 1f)
        totalTime = mediaPlayer.duration

        playbutton.setOnClickListener {
            mediaPlayer.start()
        }

        pausebutton.setOnClickListener {
            mediaPlayer.pause()
        }

        stopbutton.setOnClickListener {
            mediaPlayer.stop()
            mediaPlayer.reset()
            mediaPlayer.release()
        }

        // when user changes the seek bar the music should point to it
        seekbar.max = totalTime
        seekbar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
                if (p2) {
                    mediaPlayer.seekTo(p1)
                }
            }

            override fun onStartTrackingTouch(p0: SeekBar?) {}

            override fun onStopTrackingTouch(p0: SeekBar?) {}
        })

        // handling seekbar based on the music
        android.os.Handler().postDelayed({
            try {
                seekbar.progress = mediaPlayer.currentPosition
            } catch (exception: Exception) {
                seekbar.progress = 0
            }
        }, 1000)
    }
}
